geoh5py/io/h5_writer.py

```diff
@@ -322,17 +318,19 @@ def update_field(
 
             if entity_handle is None:
                 return
-
-            if attribute in [
-                "concatenated_attributes",
-                "metadata",
-                "options",
+            if attribute in ["concatenated_attributes", "options"]:
+                H5Writer.write_group_values(
+                    h5file, entity, attribute, compression, **kwargs
+                )
+            elif attribute in [
                 "trace_depth",
                 "values",
             ]:
                 H5Writer.write_data_values(
                     h5file, entity, attribute, compression, **kwargs
                 )
+            elif attribute == "metadata":
```

Comment:
```suggestion
            elif attribute in ["metadata", "options"]:
```
Should be handled the same way.